cluster/juju/layers/kubernetes-master/exec.d/vmware-patch/charm-pre-install

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+MY_HOSTNAME=$(hostname)
+
+if [ "${MY_HOSTNAME}" == "ubuntuguest" ]; then
+    echo "Detected broken vsphere integration. Applying hostname override"
+    if [ -z "${JUJU_UNIT_NAME}" ]; then
```

Comment:
We should do bash fallback instead of just exiting. For example

```
: ${JUJU_UNIT_NAME:=`uuidgen`}
```

At the top of the script will make sure that JUJU_UNIT_NAME exists or is assigned to a UUID